public/scripts/project/lock.js

```diff
@@ -0,0 +1,51 @@
+// Our lock system for projects is very simple. If the lock exists,
+// it's almost guaranteed to be valid since the expiry serves
+// to prevent completely locking access to a project in edge cases.
+define(function() {
+  var EXPIRY = 60000; // 1000ms * 60s
```

Comment:
Given that logic, I'd make it 10s or something, not 1 minute.  Imagine your browser crashes, you reload it 20s later, it's locked (but not open in any tabs), you refresh, you refresh, you refresh, you give up--you're still under 1 minute.   This seems like bad UX to me.  Expire early imho.